probePlugin/src/main/scala/org/virtuslab/ideprobe/handlers/RunConfigurations.scala

```diff
@@ -1,37 +1,89 @@
+import com.intellij.openapi.actionSystem.{CommonDataKeys, LangDataKeys, PlatformDataKeys}
+import com.intellij.openapi.project.{DumbService, Project}
+import com.intellij.openapi.roots.ModuleRootManager
 import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.{JavaPsiFacade, PsiManager}
+import com.intellij.testFramework.MapDataContext
+import org.virtuslab.ideprobe.Extensions._
 import org.virtuslab.ideprobe.RunnerSettingsWithProcessOutput
-import org.virtuslab.ideprobe.protocol.ApplicationRunConfiguration
-import org.virtuslab.ideprobe.protocol.JUnitRunConfiguration
-import org.virtuslab.ideprobe.protocol.ProcessResult
-import org.virtuslab.ideprobe.protocol.TestRun
-import org.virtuslab.ideprobe.protocol.TestStatus
-import org.virtuslab.ideprobe.protocol.TestSuite
-import org.virtuslab.ideprobe.protocol.TestsRunResult
+import org.virtuslab.ideprobe.protocol._
+
+import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
 import scala.concurrent.ExecutionContext
-import org.virtuslab.ideprobe.Extensions._
 
 object RunConfigurations extends IntelliJApi {
+  def execute(runConfiguration: TestRunConfiguration)(implicit ec: ExecutionContext): TestsRunResult =
+    BackgroundTasks.withAwaitNone {
+      val project = Projects.resolve(ProjectRef.Default)
+      val module = Modules.resolve(runConfiguration.module)
+
+      val dataContext = new MapDataContext
+      dataContext.put(CommonDataKeys.PROJECT, project)
+      dataContext.put(LangDataKeys.MODULE, module)
+
+      val projectPath = Paths.get(project.getBasePath)
+
+      // TODO getContentRoots returns an array, is taking head only reliable?
+      val moduleVirtualFile = ModuleRootManager.getInstance(module).getContentRoots.head
+      val moduleDir = runOnUISync {
+        PsiManager.getInstance(project).findDirectory(moduleVirtualFile)
+      }
+
+      dataContext.put(PlatformDataKeys.SELECTED_ITEMS, List(moduleDir).asJava.toArray)
+
+      val wsVirtualFile = VFS.toVirtualFile(projectPath)
+      dataContext.put(PlatformDataKeys.PROJECT_FILE_DIRECTORY, wsVirtualFile)
+
+      val location = PsiLocation.fromPsiElement(moduleDir)
+      dataContext.put(Location.DATA_KEY, location)
+
+      val configurationContext = ConfigurationContext.getFromContext(dataContext)
+      val runManager = configurationContext.getRunManager.asInstanceOf[RunManagerEx]
+      val configurationFromContext = runOnUISync(configurationContext.getConfiguration)
+
+      runManager.setTemporaryConfiguration(configurationFromContext)
+      runManager.setSelectedConfiguration(configurationFromContext)
+
+      val configurations = runOnUISync {
+        configurationContext.getConfigurationsFromContext
+      }
+      val producer = runConfiguration.runnerNameFragment match {
+        case Some(fragment) =>
+          configurations
+            .find(_.toString.toLowerCase contains fragment.toLowerCase)
```

Comment:
let's make it case sensitive
do we need substring? what are the example names?

Reply:
the examples are `JUnit: All in Hello (1) (level: WORKSPACE)` or `ScalaTest: ScalaTests in 'ws' (level: WORKSPACE)` so provider runner names like `JUnit` or `ScalaTest` could be applied in such cases